tests/org.palladiosimulator.dataflow.confidentiality.analysis.tests/src/org/palladiosimulator/dataflow/confidentiality/analysis/constraint/ConstraintResultTest.java

```diff
@@ -0,0 +1,192 @@
+        		.flatMap(it -> it.stream()).collect(Collectors.toList());
+        printNodeInformation(node);
+
+        return dataSensitivity.stream()
+            .anyMatch(l -> l.getName()
+                .equals("Personal")) && serverLocation.stream()
+                    .anyMatch(l -> l.getName()
+                        .equals("nonEU"));
+    }
+    
+    /**
+     * Indicates whether an element in an action sequence violates the constraint of the
+     * datastore test model
+     * 
+     * @param node
+     *            Element of the action sequence
+     * @return Returns true, if the constraint is violated. Otherwise, the method returns false.
+     */
+    private boolean dataStoreCondition(AbstractActionSequenceElement<?> node) {
+    	List<Literal> assignedRoles = node.getNodeCharacteristicsWithName("AssignedRole");
+    	Map<DataFlowVariable, List<Literal>> grantedRoles = node.getDataFlowCharacteristicsWithName("GrantedRole");
+    	
+        printNodeInformation(node);
+        
+        if (assignedRoles.isEmpty()) {
+        	return false;
+        }
+        
+        return !grantedRoles.entrySet().stream()
+        		.allMatch(df -> df.getValue().stream().allMatch(it -> assignedRoles.contains(it)));
+    }
+    
+    /**
+     * Indicates whether an element in an action sequence violates the constraint of the
+     * return test model
+     * 
+     * @param node
+     *            Element of the action sequence
+     * @return Returns true, if the constraint is violated. Otherwise, the method returns false.
+     */
+    private boolean returnCondition(AbstractActionSequenceElement<?> node) {
+    	List<Literal> assignedNode = node.getNodeCharacteristicsWithName("AssignedRole");
+    	List<Literal> assignedVariables = node.getDataFlowCharacteristicsWithName("AssignedRole").values().stream()
+    			.flatMap(it -> it.stream())
+    			.collect(Collectors.toList());
+    	
+        printNodeInformation(node);
+        if (assignedNode.isEmpty() || assignedVariables.isEmpty()) {
+        	return false;
+        }
+        assignedNode.removeAll(assignedVariables);
+        return !assignedNode.isEmpty();
+    }
+    
+    /**
+     * Provides Arguments for the testConstraintResults test.
+     * @return Returns a stream of Arguments containing the test model with constraint and expected results
+     */
+    private Stream<Arguments> provideTestConstraintResults() {
+    	Predicate<AbstractActionSequenceElement<?>> travelPlannerConstraint = node -> travelPlannerCondition(node);
+    	Predicate<AbstractActionSequenceElement<?>> internationalOnlineShopConstraint = node -> internationalOnlineShopCondition(node);
+
+    	StandalonePCMDataFlowConfidentialtyAnalysis mutipleAssembliesAnalysis = 
+    			super.initializeAnalysis(Paths.get("models", "MultipleAssembliesTest", "default.usagemodel"), Paths.get("models", "MultipleAssembliesTest", "default.allocation"));
```

Comment:
I'm not happy with the name `MultipleAssembliesTest`. I understand, that you allocated multiple dependencies on a single resource, but the name indicates that there are somehow multiple assemblies in the model (or something like that). However, I do not have a good name either. Maybe something like `MultipleAssembliesOneResourceTest` to clarify it?

Reply:
I have renamed it to `OneAssemblyMutlipleResourceContainerTest` to indicate what this model is testing